src/main/java/com/google/sps/servlets/DestinationDataServlet.java

```diff
@@ -0,0 +1,133 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.ArrayList;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.List;
+import java.util.Arrays;
+import java.util.stream.Collectors;
+import java.util.Set;
+import java.util.HashSet;
+
+/* Returns a destination created from user submitted information */
+@WebServlet("/destination-data")
+public class DestinationDataServlet extends HttpServlet {
+  
+  // Temporarily stores the destination created by the user
+  public List<com.google.sps.data.Destination> destinations = new ArrayList<>();
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String name = request.getParameter("name");
+
+    com.google.sps.data.LatLng location = new com.google.sps.data.LatLng.Builder()
+            .withLat(Double.parseDouble(request.getParameter("latitude")))
+            .withLng(Double.parseDouble(request.getParameter("longitude")))
+            .build();
+
+    String city = request.getParameter("city");
+    String description = request.getParameter("description");
+
+    com.google.sps.data.Riddle riddle = new com.google.sps.data.Riddle.Builder()
+            .withPuzzle(request.getParameter("riddle"))
+            .withHint(request.getParameter("hint1"))
+            .withHint(request.getParameter("hint2"))
+            .withHint(request.getParameter("hint3"))
+            .build();
+            
+    /* Retrieves the obscurity level chosen by the user as a List of Strings and converts the level to an Enum Obscurity value */
+    List<String> levels = Arrays.stream(request.getParameterValues("obscurity")).filter(level -> level != null).collect(Collectors.toList());
+    com.google.sps.data.Destination.Obscurity level = convertLevelToEnum(levels);
+
+    /* Retrieves the tags selected by the user as a List of Strings and converts them into a Set of Enum Tags */
+    List<String> tags = Arrays.stream(request.getParameterValues("tag")).filter(tag -> tag != null).collect(Collectors.toList());
+    Set<com.google.sps.data.Destination.Tag> checkedTags = convertTagsToEnum(tags);
+
+    com.google.sps.data.Destination d1 = new com.google.sps.data.Destination.Builder()
+            .withName(name)
+            .withLocation(location)
+            .withCity(city)
+            .withDescription(description)
+            .withRiddle(riddle)
+            .withTags(checkedTags)
+            .withObscurity(level)
+            .build();
+    
+    destinations.add(d1);
+    response.sendRedirect("/destination-data");
```

Comment:
Should this be redirecting to the .html page instead? From what I remember, sending a POST request to /destination-data will redirect to this URL by default.

Reply:
I redirect to /destination-data just to show that the information has been retrieved. I plan to change it to redirect to a .html once datastore is implemented.

Reply:
leave a TODO comment to indicate that you're planning on changing it to redirect to an HTML file